src/main/java/minicraft/entity/furniture/Furniture.java

```diff
@@ -52,7 +52,7 @@ public Furniture(String name, Sprite sprite, int xr, int yr) {
 	@Override
 	public Furniture clone() {
 		try {
-			return getClass().newInstance();
+			return getClass().getDeclaredConstructor().newInstance();
```

Comment:
Scary code

Reply:
How so? `getClass().newInstance()` skips compile-time checks of constructors. I would rather get a compile error (with `getClass().getDeclaredConstructor().newInstance()`) than a runtime error from this.

Reply:
Thank you for explaining